src/libpmemfile/path_resolve.c

```diff
@@ -303,9 +303,25 @@ resolve_path(struct fd_desc at,
 		if (get_stat(result, &stat_buf) != 0)
 			break;
 
+		const struct fd_association *pmem_fda = &result->at.pmem_fda;
+
 		if (!is_last_component)
 			result->path[end] = '/';
 
+		/*
+		 * If we are at "/" and the next component is ".." we have to
+		 * exit the pmemfile pool and reevaluate the path using
+		 * syscalls.
+		 */
+		if (at_root && (end - resolved) == 2 &&
+				memcmp(&result->path[resolved], "..", 2) == 0) {
+			exit_pool(result, resolved, &size);
+			at_root = false;
+			continue;
+		}
+
+		at_root = false;
```

Comment:
Why set it to false here?
What if, e.g.: it was true, and the next component is "."?